manage_employees/employee_management_services.bal

```diff
@@ -0,0 +1,173 @@
+import ballerina/http;
+import ballerina/log;
+
+type EmployeeInfo {
+    string name;
+    int age;
+    string city;
+    string employee_id;
+    string department_id;
+    int record_id;
+    string created_date;
+    string last_updated;
+};
+
+endpoint http:Listener EmployeeMgtServiceListener {
+    port:9024
+};
+
+//Endpoint for HR system
+endpoint http:Client hrEP {
+    url: "http://localhost:9023/hr/employee-mgt"
+};
+
+//Endpoint for Data Migration Services
+endpoint http:Client dataMigrationEP {
+    url: "http://localhost:9022/data-migration"
+};
+
+@http:ServiceConfig {
+    basePath: "/"
+}
+service<http:Service> manageEmployee bind EmployeeMgtServiceListener {
+
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/employee"
+    }
+    addEmployee(endpoint caller, http:Request addEmployeeRequest) {
+        EmployeeInfo employeeInfo;
+        http:Request hrSystemRequest = new;
+        json requestJson = {};
+        string employeeId;
+        string newEmployeeId;
+        string departmentId;
+        string newDepartmentId;
+
+        //Process request JSON payload and validate
+        var payloadJson = addEmployeeRequest.getJsonPayload();
```

Comment:
without assigning the json payload to a variable, why don't you have it as an argument for match.

match (addEmployeeRequest.getJsonPayload()) { ... }

Reply:
Fixed